compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -608,11 +608,12 @@ public void visit(BLangAnnotationAttachment astAnnotAttach) {
     }
 
     private boolean isCompileTimeAnnotationValue(BLangExpression expr) {
-        // TODO Compile time literal constants
         switch (expr.getKind()) {
             case LITERAL:
             case NUMERIC_LITERAL:
                 return true;
+            case INVOCATION:
+                return isCompileTimeAnnotationValue(((BLangInvocation) expr).expr);
```

Comment:
Shouldn't we pass `(BLangInvocation) expr` itself to `isCompileTimeAnnotationValue` and add a case there to check the return type?